### PR TITLE
Add Python/Java/C++ Solution to Soldier and Bananas (CF)

### PR DESCRIPTION
**Pull request by ryanchou-dev**

### CODE REVIEW

File: solutions/cf-546A.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+```
+</CPPSection>
+
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class Main {
```
Comment (SansPapyrus683) [inline solutions/cf-546A.mdx:58]:
```suggestion
public class SoldierBananas {
```

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/cf-546A.mdx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	public static void main(String[] args) throws IOException {
```
Comment (nathangong) [inline solutions/cf-546A.mdx:58]:
```suggestion
	public static void main(String[] args) {
```

File: solutions/cf-546A.mdx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		Scanner input = new Scanner(System.in);
+		int first = input.nextInt();
+		long money = input.nextInt();
+		int num = input.nextInt();
+	
+		long cost = num * (num + 1) / 2;
+		cost *= first;
+
+		System.out.println(((cost > money) ? cost - money : 0));
```
Comment (nathangong) [inline solutions/cf-546A.mdx:67]: Maybe do `Math.max(cost-money, 0)` instead?

**Review by ar492 — APPROVED** [review]

File: solutions/cf-546A.mdx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+int main() {
+	int first, num;
+	long long money;
+	cin >> first >> money >> num;
+
+	int cost = num * (num + 1) / 2;
+	cost *= first;
+
+	cout << ((cost > money) ? cost - money : 0);
```
Comment (nathangong) [inline solutions/cf-546A.mdx:46]: also do the max thing for c++ and python

File: solutions/cf-546A.mdx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+source: CF
+title: Soldier and Bananas
+author: Ryan Chou
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/18034)
+
+## Explanation
+
+We can find the total cost of $w$ bananas by getting the sum of numbers from $1 - W$ and multiplying by the price of the first banana, $k$.
```
Comment (nathangong) [inline solutions/cf-546A.mdx:12]: optional
```suggestion
We can find the total cost of $w$ bananas by calculating the sum of numbers from $1 - W$ and multiplying by the price of the first banana, $k$.
```

**Review by dongliu0426 — APPROVED** [review]

**Pull request merged**